pkg/envoy/config.go

```diff
@@ -4,26 +4,53 @@ import (
 	"fmt"
 
 	v1 "k8s.io/api/core/v1"
+
+	"github.com/openservicemesh/osm-health/pkg/kuberneteshelper"
+	"github.com/openservicemesh/osm-health/pkg/osm"
+	osmCLI "github.com/openservicemesh/osm/pkg/cli"
 )
 
-// tempStruct implements ConfigGetter and is temporary until https://github.com/openservicemesh/osm-health/pull/14 is merged
-// TODO(draychev): delete this
-type tempStruct struct {
+// ConfigGetterStruct implements ConfigGetter interface.
+type ConfigGetterStruct struct {
 	*v1.Pod
+	osm.ControllerVersion
 }
 
-// GetConfig implements ConfigGetter
-func (mcg tempStruct) GetConfig() (*Config, error) {
-	return &Config{}, nil
+// GetConfig implements ConfigGetter interface.
+func (mcg ConfigGetterStruct) GetConfig() (*Config, error) {
+	client, err := kuberneteshelper.GetKubeClient()
+	if err != nil {
+		return nil, err
+	}
+
+	config, err := kuberneteshelper.GetKubeConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	namespace := mcg.Pod.Namespace
+	podName := mcg.Pod.Name
+	localPort := osm.EnvoyAdminPort[mcg.ControllerVersion]
+	query := "config_dump?include_eds"
+	// This function becomes available in github.com/openservicemesh/osm at 9be251135819c360ce2b9cf77087c88ab1e3f54a
+	configBytes, err := osmCLI.GetEnvoyProxyConfig(client, config, namespace, podName, localPort, query)
+	log.Trace().Msgf("Envoy config for %s/%s", mcg.Pod.Namespace, mcg.Pod.Name)
```

Comment:
Perhaps `Queried envoy config for %s/%s using query %s` is better?

Reply:
I should completely delete this line.